sdk/monitor/monitor-opentelemetry-exporter/src/platform/nodejs/baseSender.ts

```diff
@@ -155,6 +155,11 @@ export abstract class BaseSender {
       } else if (restError.statusCode && isRetriable(restError.statusCode)) {
         this.networkStatsbeatMetrics?.countRetry(restError.statusCode);
         return this.persist(envelopes);
+      } else if (restError.statusCode === 400 && restError.message.includes("Invalid instrumentation key")) {
+        const invalidInstrumentationKeyError = new Error("Invalid instrumentation key");
+        this.incrementStatsbeatFailure();
```

Comment:
Instrumentation key cannot be updated, so in this case you will fail multiple times before shutting down Statbeat, in theory you should turn it off immediately here, is fine like this if it require to many changes